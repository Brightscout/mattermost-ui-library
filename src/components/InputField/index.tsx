import React from 'react';
import {FormControl} from 'react-bootstrap';

type InputFieldProps = {
    type?: 'text' | 'password' | 'email';
    label?: string | JSX.Element;
    placeholder?: string;
    value?: string;
    onChange?: (e: React.ChangeEvent<HTMLInputElement>) => void;
    error?: boolean | string;
    disabled?: boolean;
    className?: string;
}

const Input = ({
    type = 'text',
    label,
    placeholder = '',
    value = '',
    onChange,
    error,
    disabled = false,
    className = '',
}: InputFieldProps) => (
    <div className={`form-group ${className}`}>
        {label && <label className='form-group__label wt-400'>{label}</label>}
        <FormControl
            type={type}
            value={value}
            onChange={(e) => onChange?.(e as React.ChangeEvent<HTMLInputElement>)}
            placeholder={placeholder}
            disabled={disabled}
            className={`form-group__control border-radius-4 ${error && 'form-group__control--err error-text'}`}
        />
        {(error && typeof error === 'string') && <p className='form-group__err-text error-text font-size-14 margin-top-5'>{error}</p>}
    </div>
);

export default Input;
