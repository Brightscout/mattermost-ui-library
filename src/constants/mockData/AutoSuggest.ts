// Mock data for "AutoSuggest" component tests
export const onInputValueChange = jest.fn();
export const onOptionClick = jest.fn();
export const suggestionConfig = {
    suggestions: [{
        label: 'Label 1',
        value: 'Value 1',
    }, {
        label: 'Label 2',
        value: 'Value 2',
    }, {
        label: 'Label 3',
        value: 'Value 3',
    }],
    renderValue: (suggestion: Record<string, string>) => suggestion.label,
};
