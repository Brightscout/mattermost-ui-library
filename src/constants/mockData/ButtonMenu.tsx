import React from 'react';

// Mock data for "ButtonMenu" component tests
export const Icon = <i className='icon icon-globe' />;
export const onBtnClick = jest.fn();
export const buttons = [{
    text: 'Button Text',
    icon: Icon,
    onClick: onBtnClick,
}, {
    text: 'Button Text 2',
    icon: Icon,
    onClick: onBtnClick,
}];
