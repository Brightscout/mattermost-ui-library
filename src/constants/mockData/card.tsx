import React from 'react';

// Mock data for "card" related component tests

// Mock data for "BaseCard"
export const BaseCardChildren = <div>{'Base card children'}</div>;

// Mock data for "SubscriptionCard"
export const editHandler = jest.fn();
export const deleteHandler = jest.fn();
export const subscriptionCardBody = {
    list: ['List Item 1', 'List Item 2', 'List Item 3'],
    labelValuePairs: [
        {label: 'Label 1', value: 'Value 1'},
        {label: 'Label 2', value: 'Value 2'},
        {label: 'Label 3', value: 'Value 3'},
        {label: 'Label 4', value: 'Value 4'},
    ],
};
// Mock Icons
export const EditIcon = <i className='fa fa-edit' />;
export const DeleteIcon = <i className='fa fa-delete' />;
// Mock channel data
export const channelData = {
    create_at: 19304124124,
    creator_id: 'klsioer239042',
    delete_at: 19304124124,
    display_name: 'Channel Display Name',
    extra_update_at: 19304124124,
    group_constrained: null,
    header: 'Channel Header',
    id: 'jw390r23iofs9032',
    last_post_at: 19304124124,
    last_root_post_at: 19304124124,
    name: 'Channel Name',
    policy_id: null,
    props: null,
    purpose: 'Channel Purpose',
    scheme_id: null,
    shared: null,
    team_display_name: 'Team Display Name',
    team_id: 'ior9823ijoiewro3w',
    team_name: 'Team Name',
    team_update_at: 19304124124,
    total_msg_count: 1,
    total_msg_count_root: 1,
    type: 'P',
    update_at: 19304124124,
};
export const subscriptionCardProps = {
    header: 'Subscription Card Header',
    label: 'Label',
    cardBody: subscriptionCardBody,
    description: 'Description',
    channel: channelData,
    editIcon: EditIcon,
    deleteIcon: DeleteIcon,
    onEdit: editHandler,
    onDelete: deleteHandler,
    className: 'subscription-card-custom-class',
};

// Mock data for "Card" component
export const cardBody = [{
    sectionHeading: 'Section Heading',
    data: [{
        icon: <i className='fa fa-edit' />,
        label: 'Label 1',
    }, {
        icon: <i className='fa fa-link' />,
        label: 'Label 2',
    }],
}];
export const cardProps = {
    cardHeader: 'Card Header',
    cardSubHeader: 'Card sub header',
    cardBody,
    onEdit: editHandler,
    onDelete: deleteHandler,
    editIcon: EditIcon,
    deleteIcon: DeleteIcon,
    className: 'card-custom-class',
};
