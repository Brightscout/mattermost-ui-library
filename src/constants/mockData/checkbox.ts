// Mock data for "Checkbox" component
export const checkboxLabel = 'Checkbox Label';
export const checkboxChangeHandler = jest.fn();
export const changeHandler = jest.fn();
