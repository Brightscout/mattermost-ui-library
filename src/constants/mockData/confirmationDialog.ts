
// Mock data for "ConfirmationDialog" component
export const hideHandler = jest.fn();
export const confirmHandler = jest.fn();
export const confirmationDialogProps = {
    title: 'Dialog Title',
    confirmationMsg: 'Dialog Message',
    show: true,
    onHide: hideHandler,
    loading: true,
    onConfirm: confirmHandler,
    className: 'confirmation-dialog-class',
    error: 'Dialog error',
};
