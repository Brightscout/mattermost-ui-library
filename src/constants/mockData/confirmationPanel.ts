// Mock data for "ConfirmationPanel" component
export const hideHandler = jest.fn();
export const confirmHandler = jest.fn();
export const confirmationPanelProps = {
    confirmationMsg: 'Dialog Message',
    onHide: hideHandler,
    onConfirm: confirmHandler,
    loading: true,
    className: 'confirmation-dialog-class',
    error: 'Dialog error',
};
