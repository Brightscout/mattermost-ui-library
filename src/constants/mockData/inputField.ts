// Mock data for "InputField" component
export const inputChangeHandler = jest.fn();
export const inputFieldProps = {
    label: 'Input Label',
    placeholder: 'Input Placeholder',
    value: 'InputValue',
    className: 'input-field-class',
    disabled: false,
    error: 'Input field error',
    onChange: inputChangeHandler,
};
