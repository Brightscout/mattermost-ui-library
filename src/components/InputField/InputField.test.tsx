import React from 'react';
import {shallow, ShallowWrapper} from 'enzyme';

// Mock data
import {inputFieldProps} from '@Constants/mockData/inputField';

import InputField from '.';

describe('InputField', () => {
    let component: ShallowWrapper<any, Readonly<{}>, React.Component<{}, {}, any>>;

    beforeEach(() => {
        component = shallow(
            <InputField
                type='text'
                {...inputFieldProps}
            />,
        );
    });

    it('Should render correctly', () => {
        expect(component).toMatchSnapshot();
    });

    it('Should apply the passed "className" prop', () => {
        expect(component.hasClass(inputFieldProps.className)).toBeTruthy();
    });

    it('Should render the label correctly', () => {
        expect(component.text().includes(inputFieldProps.label)).toBeTruthy();
    });

    it('Should render the error correctly', () => {
        expect(component.find('.form-group__err-text').text().includes(inputFieldProps.error)).toBeTruthy();
    });
});
