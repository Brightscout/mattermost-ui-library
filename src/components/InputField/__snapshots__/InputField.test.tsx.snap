// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputField Should render correctly 1`] = `
<Input
  className="input-field-class"
  disabled={false}
  error="Input field error"
  label="Input Label"
  onChange={[MockFunction]}
  placeholder="Input Placeholder"
  type="text"
  value="InputValue"
>
  <div
    className="form-group input-field-class"
  >
    <label
      className="form-group__label wt-400"
    >
      Input Label
    </label>
    <FormControl
      className="form-group__control border-radius-4 form-group__control--err error-text"
      disabled={false}
      onChange={[Function]}
      placeholder="Input Placeholder"
      type="text"
      value="InputValue"
    >
      <input
        className="form-group__control border-radius-4 form-group__control--err error-text form-control"
        disabled={false}
        onChange={[Function]}
        placeholder="Input Placeholder"
        type="text"
        value="InputValue"
      />
    </FormControl>
    <p
      className="form-group__err-text error-text font-14 margin-top-5"
    >
      Input field error
    </p>
  </div>
</Input>
`;
